File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int l = 1, r = 1;
        while (r < nums.size())
        {
            if (nums[r] != nums[r - 1])
            {
                nums[l] = nums[r];
                l++;
            }
            r++;
        }

        return l;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0

        for n in nums:
            if i < 1 or n > nums[i - 1]: # for any number of duplicates allowed we can generalize it as if i < k or nums[i - k] where k is the number of duplicates allowed 
                nums[i] = n
                i += 1
        return i